pkg/plugin/processor/builtin/setField.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type setField struct {
+	referenceResolver sdk.ReferenceResolver
+	value             string
+
+	sdk.UnimplementedProcessor
+}
+
+func (p *setField) Specification() (sdk.Specification, error) {
```

Comment:
Should we also register all the rewritten plug-ins? I did so with the JS processor: https://github.com/ConduitIO/conduit/pull/1374/files#diff-6c82ac75684bca1ba4a7e1c33a0ac2058bed4b91712149bd8d1806b07e0904e6. Or maybe there's a reason we're not doing that yet?

Reply:
We can prepare that default processor map in the JS processor PR and then add the others in a separate PR, as to not get merge conflicts.

Reply:
hmm I was gonna add them at the end, but no reason not to.
however, since we're gonna need `processor.New` for each processor, does that mean I need to separate each processor to their own directory? cuz we can't redeclare `New` multiple times in the same package.

Reply:
I can't comment much on how Go idiomatic is it to have a package for each of them. Personally, I feel that might be a bit too much since some processors will be really simple. Maybe we can include the processor name in the constructor name? We'd then have `builtin.NewRenameField` for example.

Reply:
the specific processor name constructor sounds like a good idea, but let's still revisit this after JS processor is merged and we have the processor map ready